test/e2e/multiple_namespaces_test.go

```diff
@@ -0,0 +1,160 @@
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("[Basic] Backup/restore of 2 namespaces", func() {
+
+	client, err := newTestClient()
+	Expect(err).To(Succeed(), "Failed to instantiate cluster client for multiple namespace tests")
+
+	BeforeEach(func() {
+		var err error
+		flag.Parse()
+		uuidgen, err = uuid.NewRandom()
+		Expect(err).To(Succeed())
+		if installVelero {
+			Expect(veleroInstall(context.Background(), veleroImage, veleroNamespace, cloudProvider, objectStoreProvider, false,
+				cloudCredentialsFile, bslBucket, bslPrefix, bslConfig, vslConfig, "")).To(Succeed())
+
+		}
+	})
+
+	AfterEach(func() {
+		if installVelero {
+			timeoutCTX, _ := context.WithTimeout(context.Background(), time.Minute)
+			err := veleroUninstall(timeoutCTX, client.kubebuilder, installVelero, veleroNamespace)
+			Expect(err).To(Succeed())
+		}
+
+	})
+
+	Context("When I create 2 namespaces", func() {
+		It("should be successfully backed up and restored", func() {
+			backupName := "backup-" + uuidgen.String()
+			restoreName := "restore-" + uuidgen.String()
+			fiveMinTimeout, _ := context.WithTimeout(context.Background(), 5*time.Minute)
+			RunMultipleNamespaceTest(fiveMinTimeout, client, "nstest-"+uuidgen.String(), 2,
+				backupName, restoreName)
+		})
+	})
+})
+
+var _ = Describe("[Scale] Backup/restore of 2500 namespaces", func() {
+
+	client, err := newTestClient()
+	Expect(err).To(Succeed(), "Failed to instantiate cluster client for multiple namespace tests")
+
+	BeforeEach(func() {
+		var err error
+		flag.Parse()
+		uuidgen, err = uuid.NewRandom()
+		Expect(err).To(Succeed())
+		if installVelero {
+			Expect(veleroInstall(context.Background(), veleroImage, veleroNamespace, cloudProvider, objectStoreProvider, false,
+				cloudCredentialsFile, bslBucket, bslPrefix, bslConfig, vslConfig, "")).To(Succeed())
+
+		}
+	})
+
+	AfterEach(func() {
+		if installVelero {
+			timeoutCTX, _ := context.WithTimeout(context.Background(), time.Minute)
+			err := veleroUninstall(timeoutCTX, client.kubebuilder, installVelero, veleroNamespace)
+			Expect(err).To(Succeed())
+		}
+
+	})
+
+	Context("When I create 2500 namespaces", func() {
+		It("should be successfully backed up and restored", func() {
+			backupName := "backup-" + uuidgen.String()
+			restoreName := "restore-" + uuidgen.String()
+			oneHourTimeout, _ := context.WithTimeout(context.Background(), 1*time.Hour)
+			RunMultipleNamespaceTest(oneHourTimeout, client, "nstest-"+uuidgen.String(), 2500,
+				backupName, restoreName)
+		})
+	})
+})
+
+func RunMultipleNamespaceTest(ctx context.Context, client testClient, nsBaseName string, numberOfNamespaces int, backupName string, restoreName string) error {
+	shortTimeout, _ := context.WithTimeout(ctx, 5*time.Minute)
+	defer cleanupNamespaces(ctx, client, nsBaseName) // Run at exit for final cleanup
```

Comment:
If `cleanupNamespaces` returns an error, it will be ignored due to the `defer` here. I think it's still better to call the cleanup using `defer` as that will ensure that we attempt it even if the test fails mid way through.

Does it matter if the error is ignored? I don't think we should mark the test as failed if it couldn't clean up the namespaces, but maybe we could log the error doing something like the following:
```
defer func() {
  if err := cleanupNamespaces(ctx, client, nsBaseName); err != nil {
    fmt.Println("Error during namespace cleanup:", err)
  }
}
```